### PR TITLE
Add BigQuery job link

### PR DESCRIPTION
**Pull request by nakamura1878**
This pull request introduces a new helper class for constructing BigQuery Job Detail Links and integrates it into the existing BigQuery operators and links in the Google Cloud provider for Airflow. The most important changes include the addition of the `BigQueryJobDetailLink` class, its integration into the `BigQueryInsertJobOperator`, and the necessary imports and configurations.

New feature addition:

* Added `BIGQUERY_JOB_DETAIL_LINK` constant to construct URLs for BigQuery Job Detail Links in `airflow/providers/google/cloud/links/bigquery.py`.
* Created `BigQueryJobDetailLink` class to handle the construction and persistence of BigQuery Job Detail Links in `airflow/providers/google/cloud/links/bigquery.py`.

Integration into existing operators:

* Imported `BigQueryJobDetailLink` in `airflow/providers/google/cloud/operators/bigquery.py` to make it available for use in BigQuery operators.
* Updated `operator_extra_links` in `BigQueryInsertJobOperator` to include `BigQueryJobDetailLink` in `airflow/providers/google/cloud/operators/bigquery.py`.
* Added logic to persist BigQuery Job Detail Links in the `execute` method of `BigQueryInsertJobOperator` in `airflow/providers/google/cloud/operators/bigquery.py`.

Configuration update:

* Added `BigQueryJobDetailLink` to the `extra-links` section in `airflow/providers/google/provider.yaml` to ensure it is recognized as an extra link.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/google/cloud/links/bigquery.py [hunk 1]
```diff
@@ -35,6 +35,9 @@
     BIGQUERY_BASE_LINK
     + "?referrer=search&project={project_id}&d={dataset_id}&p={project_id}&page=table&t={table_id}"
 )
+BIGQUERY_JOB_DETAIL_LINK = (
+    BIGQUERY_BASE_LINK + "?project={project_id}&ws=!1m5!1m4!1m3!1s{project_id}!2s{job_id}!3s{location}"
```
Comment (shahar1) [inline airflow/providers/google/cloud/links/bigquery.py:39]:
```suggestion
    BIGQUERY_BASE_LINK + "?project={project_id}&ws=!1m5!1m4!1m3!1s{project_id}!2s{job_id}!3s{location}"
```

Are the "!1mX!" strings necessary?

**Review by shahar1 — COMMENTED** [review]
Welcome to Apache Airflow and thank you for your first contribution!
Overall LGTM, I've got a small question though :)

File: airflow/providers/google/cloud/links/bigquery.py (hunk 1 above)
Comment (nakamura1878) [reply airflow/providers/google/cloud/links/bigquery.py:39]: @shahar1 
Thank you for the review! Without that string, I couldn't open the job result.

**Review by shahar1 — APPROVED** [review]

**Comment by shahar1** [comment]
@nakamura1878  Woops, only now I noticed that this PR was targeted at `v2-10-test` - PRs for providers should target the `main` branch.
I'll take care of reverting this PR on `v2-10-test` and porting it to `main`.

**Comment by potiuk** [comment]
> @nakamura1878 Woops, only now I noticed that this PR was targeted at v2-10-test - PRs for providers should target the main branch.
I'll take care of reverting this PR on v2-10-test and porting it to main.

Yep - we could do it, but we could also bring it to main (forward-port :) ) without reverting - there is no big harm in getting it in.

**Comment by jscheffl** [comment]
I also just noticed it was on the "wrong" branch - but only because I had a test on v2-10-test running that was cancelled due to the merge.

**Comment by shahar1** [comment]
> I also just noticed it was on the "wrong" branch - but only because I had a test on v2-10-test running that was cancelled due to the merge.

Apologies for that :)
@potiuk fair enough. Is there any stance about removing the providers from `v2-10-test` to avoid such confusion? I assume that it hasn't been done until now for a reason, not sure what though if we continue maintaining them in `main`.

**Comment by jscheffl** [comment]
> > I also just noticed it was on the "wrong" branch - but only because I had a test on v2-10-test running that was cancelled due to the merge.
> 
> Apologies for that :) @potiuk fair enough. Is there any stance about removing the providers from `v2-10-test` to avoid such confusion? I assume that it hasn't been done until now for a reason, not sure what though considering that we continue maintaining them in `main`.

No we just keep them here. Would be "too much" cleanup and CI has still a lot of dependencies and cross-checks.

**Comment by potiuk** [comment]
> > > I also just noticed it was on the "wrong" branch - but only because I had a test on v2-10-test running that was cancelled due to the merge.
> > 
> > 
> > Apologies for that :) @potiuk fair enough. Is there any stance about removing the providers from `v2-10-test` to avoid such confusion? I assume that it hasn't been done until now for a reason, not sure what though considering that we continue maintaining them in `main`.
> 
> No we just keep them here. Would be "too much" cleanup and CI has still a lot of dependencies and cross-checks.

Actually it could be possible eventually. We have been removing all the "remnants" of provider tests from CI and I think with some future refactoring of providers into separate projects (working on it) - it could be possible eventually. I might attempt it as part of https://github.com/apache/airflow/issues/42632 and after https://github.com/apache/airflow/issues/44511 is completed. 

For now the remaining parts are: 
* "airflow" contains some tests that are expecting providers (executors and few others) as mentioned by @jscheffl 
* airflow "editable" build expects the "extras" to use provider's dependencies not providers itself (but that could be likely mitigated easily - and we could use the "released" providers in `v2-10-test`

It's actually quite likely that for Airflow 3 we will have "non-providers" code only in `v3-0-test`.

**Pull request merged**